### PR TITLE
Add test for installing installonly through provides

### PR DESCRIPTION
**Pull request by kontura**
For: https://issues.redhat.com/browse/RHEL-1253
(https://bugzilla.redhat.com/show_bug.cgi?id=2163474)

Requires: https://github.com/rpm-software-management/libdnf/pull/1625

### CODE REVIEW

File: dnf-behave-tests/fixtures/specs/kernel/kernel-2.0.0-1.fc29.spec [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+Epoch:          0
+Version:        2.0.0
+Release:        1.fc29
+
+License:        GPLv2 and Redistributable, no modification permitted
+URL:            https://www.kernel.org/
+
+Summary:        The Linux kernel
+
+Provides:       kernel = %{version}-%{release}
```
Comment (j-mracek) [inline dnf-behave-tests/fixtures/specs/kernel/kernel-2.0.0-1.fc29.spec:11]: May I ask you why you have explicit provide of kernel?

File: dnf-behave-tests/fixtures/specs/kernel/kernel-3.0.0-1.fc29.spec [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+%package core
+Summary:        The Linux kernel
+
+Provides:       installonlypkg(kernel)
+Provides:       kernel = %{version}-%{release}
+Provides:       kernel-uname-r = %{version}-%{release}.x86_64
+Provides:       kernel-core-uname-r = %{version}-%{release}.x86_64
+Provides:       kernel-drm-nouveau = 16
+Provides:       kernel-core = %{version}-%{release}
```
Comment (j-mracek) [inline dnf-behave-tests/fixtures/specs/kernel/kernel-3.0.0-1.fc29.spec:28]: I believe that ` kernel-core = %{version}-%{release}` is redundant.

File: dnf-behave-tests/fixtures/specs/kernel/kernel-3.0.0-1.fc29.spec [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Linux operating system.  The kernel handles the basic functions
+of the operating system: memory allocation, process allocation, device
+input and output, etc.
+
+%package modules
+Summary:        kernel modules to match the core kernel
+
+Provides:       installonlypkg(kernel-module)
+Provides:       kernel-modules-uname-r = %{version}-%{release}.x86_64
+Provides:       kernel-modules = %{version}-%{release}
```
Comment (j-mracek) [inline dnf-behave-tests/fixtures/specs/kernel/kernel-3.0.0-1.fc29.spec:43]: I believe that `Provides:       kernel-modules = %{version}-%{release}` is redundant.

**Review by j-mracek — CHANGES_REQUESTED** [review]
The test looks good, but I believe that in SPEC definitions there are redundant provides. Those provides are automatically generated. May I ask you to verify whether my suggestion is correct and if so, please remove it (I did not mark all redundancy)?

File: dnf-behave-tests/fixtures/specs/kernel/kernel-2.0.0-1.fc29.spec (hunk 1 above)
Comment (kontura) [reply dnf-behave-tests/fixtures/specs/kernel/kernel-2.0.0-1.fc29.spec]: It was present in our already existing kernel specs, mostly I just copied it.
I have removed it.

**Comment by kontura** [comment]
> The test looks good, but I believe that in SPEC definitions there are redundant provides. Those provides are automatically generated. May I ask you to verify whether my suggestion is correct and if so, please remove it (I did not mark all redundancy)?

Ok, I have removed all of the provides that I think were not needed for the tests.
Please take a look again.

**Review by j-mracek — APPROVED** [review]

**Comment by j-mracek** [comment]
LGTM

**Comment by j-mracek** [comment]
@kontura Thank you very much for fixing a complex issue

**Pull request merged**